### PR TITLE
Add gate to install CSI proxy

### PR DESCRIPTION
**Pull request by jingxu97**
Add a gate ENABLE_CSI_PROXY to enable/disable csi proxy installation.
The default value is true

**What type of PR is this?**
/kind clearup

**What this PR does / why we need it**:
To allow user to enable or disable csi proxy installation as needed
**Which issue(s) this PR fixes**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by jingxu97** [comment]
cc @msau42

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -521,3 +521,6 @@ export GCE_PRIVATE_CLUSTER_PORTS_PER_VM="${KUBE_GCE_PRIVATE_CLUSTER_PORTS_PER_VM
 # Optional: Create apiserver konnectivity server and agent.
 export ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE="${KUBE_ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE:-false}"
 export KONNECTIVITY_SERVICE_PROXY_PROTOCOL_MODE="${KUBE_KONNECTIVITY_SERVICE_PROXY_PROTOCOL_MODE:-grpc}"
+
+# Optoinal: Enable Windows CSI-Proxy
```
Comment (msau42) [inline cluster/gce/config-default.sh:525]: typo: optional

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -1530,6 +1530,7 @@ NODE_BINARY_TAR_URL: $(yaml-quote "${NODE_BINARY_TAR_URL}")
 NODE_BINARY_TAR_HASH: $(yaml-quote "${NODE_BINARY_TAR_HASH}")
 CSI_PROXY_STORAGE_PATH: $(yaml-quote "${CSI_PROXY_STORAGE_PATH}")
 CSI_PROXY_VERSION: $(yaml-quote "${CSI_PROXY_VERSION}")
+ENABLE_CSI_PROXY: $(yaml-quoote "${ENABLE_CSI_PROXY}"
```
Comment (msau42) [inline cluster/gce/util.sh:1533]: typo: quote

**Review by msau42 — COMMENTED** [review]
generally lgtm modulo typos

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (jingxu97) [reply cluster/gce/config-default.sh:525]: en, my keyboard has issues these day. thanks for checking those bad typoos.... :(

**Comment by jingxu97** [comment]
cc @pjh @yliaog

File: cluster/gce/config-default.sh [hunk 3]
```diff
@@ -521,3 +521,6 @@ export GCE_PRIVATE_CLUSTER_PORTS_PER_VM="${KUBE_GCE_PRIVATE_CLUSTER_PORTS_PER_VM
 # Optional: Create apiserver konnectivity server and agent.
 export ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE="${KUBE_ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE:-false}"
 export KONNECTIVITY_SERVICE_PROXY_PROTOCOL_MODE="${KUBE_KONNECTIVITY_SERVICE_PROXY_PROTOCOL_MODE:-grpc}"
+
+# Optional: Enable Windows CSI-Proxy
+ENABLE_CSI_PROXY="${ENABLE_CSI_PROXY:-true}"
```
Comment (yliaog) [inline cluster/gce/config-default.sh:526]: why is there no "export"?

File: cluster/gce/config-test.sh [hunk 4]
```diff
@@ -568,3 +568,6 @@ export GCE_PRIVATE_CLUSTER_PORTS_PER_VM=${KUBE_GCE_PRIVATE_CLUSTER_PORTS_PER_VM:
 export ETCD_LISTEN_CLIENT_IP=0.0.0.0
 
 export GCE_UPLOAD_KUBCONFIG_TO_MASTER_METADATA=true
+
+# Optoinal: Enable Windows CSI-Proxy
+ENABLE_CSI_PROXY="${ENABLE_CSI_PROXY:-true}"
```
Comment (yliaog) [inline cluster/gce/config-test.sh:573]: why is there no "export"?

File: cluster/gce/config-default.sh (hunk 3 above)
Comment (jingxu97) [reply cluster/gce/config-default.sh:526]: I see there are some variable using export, some don't e.g., WINDOWS_NODE_TAINTS="${WINDOWS_NODE_TAINTS:-node.kubernetes.io/os=win1809:NoSchedule}"
 I am actually not sure in which condition it should use it.

Comment (jingxu97) [reply cluster/gce/config-default.sh:526]: I think in our script setup, since there is no new process spawned, export seems not necessary/

Comment (yliaog) [reply cluster/gce/config-default.sh:526]: i guess one way to tell is to bring up a cluster, see if the env you added is passed properly. that said, i think it does not hurt to add 'export'.

Comment (jingxu97) [reply cluster/gce/config-default.sh]: I tried it out, and it is working fine. But agree with you, no harm to add export. So added it.PTAL

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 5]
```diff
@@ -403,26 +404,29 @@ function DownloadAndInstall-KubernetesBinaries {
...
-  if (ShouldWrite-File ${env:NODE_DIR}\csi-proxy.exe) {
-    $tmp_dir = 'C:\k8s_tmp'
-    New-Item -Force -ItemType 'directory' $tmp_dir | Out-Null
-    $filename = 'csi-proxy.exe'
-    $urls = "${env:CSI_PROXY_STORAGE_PATH}/${env:CSI_PROXY_VERSION}/$filename"
-    MustDownload-File -OutFile $tmp_dir\$filename -URLs $urls
-    Move-Item -Force $tmp_dir\$filename ${env:NODE_DIR}\$filename
-    # Clean up the temporary directory
-    Remove-Item -Force -Recurse $tmp_dir
+  if ("${env:ENABLE_CSI_PROXY}" -eq "true") {
```
Comment (pjh) [inline cluster/gce/windows/k8s-node-setup.psm1:407]: You could combine these two if statements together if you'd like.

File: cluster/gce/config-test.sh (hunk 4 above)
Comment (pjh) [reply cluster/gce/config-test.sh:573]: Also wondering. Why not make this look like the variables just above?

**Comment by jingxu97** [comment]
cc @mtaufen

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (jingxu97) [reply cluster/gce/config-default.sh]: fixed

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 5 above)
Comment (jingxu97) [reply cluster/gce/windows/k8s-node-setup.psm1:407]: feel it would be a little bit long, so I will leave it for now.

**Pull request merged**